aep/general/0213/aep.md.j2

```diff
@@ -1,5 +1,212 @@
 # Common components
 
-**Note:** This AEP has not yet been adopted. See
-[this GitHub issue](https://github.com/aep-dev/aep.dev/issues/52) for more
-information.
+In general, APIs should be designed to be self-contained. APIs generally need
```

Comment:
What does it mean to be self-contained? I think it suggests that you must consider carefully how you manage the protos and leverage shared definitions. But OpenAPI is more relaxed, and it is possible that your document is a description not a definition. 

What if we refocused this AEP to describe how to use shared components across multiple APIs? Something like:
- When might you wish to share responsibility for describing or defining an API across multiple documents?
- What processes might you need to ensure that any upstream changes make their way to production, documentation, and other downstream artifacts?
- What special review processes may apply in these shared responsibility situations?

Reply:
Well...I understand that OpenAPI is often used for descriptions, but is that true in the subcase where it's used for an AEP-compliant API?  Does it make sense to have an AEP-compliant API that isn't API-first?

Reply:
I don't understand the distinction between "definitions" and "descriptions", and if we need to make a distinction then we should explain the distinction here for the benefit of readers.

But I _do_ agree with @earth2marsh 's suggestion:
> What if we refocused this AEP to describe how to use shared components across multiple APIs?

I think reuse is generally a virtue when done correctly, so what we need to do here is provide guidelines for "correct reuse".

In particular, should we recommend that all common components be versioned? We have this practice and Microsoft and it is common elsewhere, e.g. the [FHIR standard].

[FHIR standard]: http://hl7.org/fhir/downloads.html

Reply:
I think the distinction is about the source of truth, and Marsh's point (IIUC) was that OAS schemas are often derived from code, rather than being defined and then used as a contract to which implementation must conform.

I don't really understand the suggestion about refocusing.  The point of this AEP is primarily to define a set of actual common components that we expect all AEP-compliant APIs to use.

Reply:
Marking resolved to get the PR in (now that it's approved), but happy to discuss this further.